runtime/metricsview/executor_rewrite_time.go

```diff
@@ -41,6 +73,47 @@ func (e *Executor) resolveTimeRange(ctx context.Context, tr *TimeRange, tz *time
 		return nil
 	}
 
+	if tr.RillTime == "" {
+		return e.resolveTimeRangeLegacy(ctx, tr, tz, executionTime)
+	}
+
+	rt, err := rilltime.Parse(tr.RillTime)
+	if err != nil {
+		return err
+	}
+
+	t, err := e.loadWatermark(ctx, executionTime)
+	if err != nil {
+		return err
+	}
+
+	minTime, err := e.GetMinTime(ctx, e.metricsView.TimeDimension)
+	if err != nil {
+		return err
+	}
```

Comment:
It feels a bit heavy to run two queries to obtain these values, and if the watermark is not equal to the max time, it will need to run three queries. Since I believe most expressions can be resolved using only one constant, can these be lazily loaded instead of eagerly?

Reply:
I think lazy load will make it a little complex. Will do a follow up that will use metrics_time_range resolver so that these things are cached.